include/arch/arc/v2/linker.ld

```diff
@@ -1,221 +1,12 @@
 /*
- * Copyright (c) 2014-2015 Wind River Systems, Inc.
+ * Copyright (c) 2020 Synopsys.
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 
-/**
- * @brief Common parts of the linker scripts for the ARCv2/EM targets.
- */
-
-#include <linker/sections.h>
-#include <linker/linker-defs.h>
-#include <linker/linker-tool.h>
-
-/* physical address of RAM */
-#ifdef CONFIG_HARVARD
-	#define ROMABLE_REGION ICCM
-	#define RAMABLE_REGION DCCM
-#else
-	#if defined(CONFIG_XIP) && (FLASH_SIZE != 0)
-		#define ROMABLE_REGION FLASH
-		#define RAMABLE_REGION SRAM
-	#else
-		#define ROMABLE_REGION SRAM
-		#define RAMABLE_REGION SRAM
-	#endif
-#endif
-
-#ifdef CONFIG_ARC_MPU_ENABLE
-	#if CONFIG_ARC_MPU_VER == 2
-		#define MPU_MIN_SIZE 2048
-	#elif CONFIG_ARC_MPU_VER == 3
-		#define MPU_MIN_SIZE 32
-	#endif
-	#define MPU_MIN_SIZE_ALIGN . = ALIGN(MPU_MIN_SIZE );
-	#if defined(CONFIG_MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT)
-		#define MPU_ALIGN(region_size) \
-			. = ALIGN(MPU_MIN_SIZE); \
-			. = ALIGN( 1 << LOG2CEIL(region_size))
-	#else
-		#define MPU_ALIGN(region_size) \
-			. = ALIGN(MPU_MIN_SIZE)
-	#endif
-#else
-	#define MPU_MIN_SIZE_ALIGN
-	#define MPU_ALIGN(region_size) . = ALIGN(4)
-#endif
 
-#if defined(CONFIG_XIP)
-	#define _DATA_IN_ROM __data_rom_start
+#if defined(__MWDT_LINKER_CMD__)
```

Comment:
this is going to be a recipe of issues, where someone changes for one and does not update the other because we do not have mwdt in CI... There needs to be one place for linker data, but I guess we have no way around this right now.

Reply:
@nashif  agree, and also commented here:
https://github.com/zephyrproject-rtos/zephyr/pull/22668#discussion_r483579164

> It seems that linker-gnu.ld and linker-mwdt.ld does have much in common.
> We should try to see if they could be aligned, but it can go in for now.

answer:
> Yep, I'd better keep it as a separate enhancement.